project.yaml

```diff
@@ -907,14 +907,43 @@ Layer:
               ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
               (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
             ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+          FROM (
+          SELECT 
+              way, highway, railway, aeroway, z_order, way_area FROM planet_osm_polygon
+            WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+              OR railway IN ('platform')
+              OR aeroway IN ('taxiway', 'helipad')
+          UNION ALL
+          SELECT -- runways with width either tagged or estimated from runway length
+              ST_Buffer(
+                way,
+                0.5*(St_Distance(ST_StartPoint(way), ST_EndPoint(way))/St_Distance_Sphere(ST_Transform(ST_StartPoint(way), 4326), ST_Transform(ST_EndPoint(way), 4326)))*
+                  CASE
+                    WHEN width ~ '^-?\d{1,4}(\.\d+)?$' THEN width::NUMERIC
```

Comment:
This entire section isn't the clearest, can you rewrite the logic?

Right now it's 0.5 * ST_Distance/St_Distance_Sphere * (width or St_Distance_Sphere/50 bounded between 12 and 75)

It might be better to do

```sql
0.5 * cos(ST_y(st_transform(st_centroid(way),4326))) * -- Correct for Mercator projection
CASE
  WHEN width... THEN width::float
  ELSE LEAST(GREATEST(ST_Length(way::geography)/50,12.0),75.0)
END
```